### PR TITLE
Modified ingress config (#130)

### PR DESCRIPTION
**Pull request by D-Sai-Venkatesh**
## Why are these changes needed?
Added a new ingress controller to enable both KFP and Minio to take separate default URL paths.

## Related issue number (if any).
130

### CODE REVIEW

**Review by blublinsky — APPROVED** [review]

File: kind/README.md [hunk 1]
```diff
@@ -93,7 +93,7 @@ kubectl wait --for=condition=ready --all pod -n kuberay --timeout=120s
 
 To access the API server and Kubeflow pipeline UI externally, we make use NGINX ingress.
 
-Install [Ingress NGNIX](https://kind.sigs.k8s.io/docs/user/ingress/#ingress-nginx) and wait for it to be ready:
+Install [Ingress NGNIX](https://kind.sigs.k8s.io/docs/user/ingress/#ingress-nginx) for KFP and RAY and wait for it to be ready:
```
Comment (roytman) [inline kind/README.md:96]: `for KFP and Ray/MinIO`

File: kind/hack/nginx_deploy_minio.yaml [hunk 2]
```diff
@@ -0,0 +1,673 @@
+apiVersion: v1
```
Comment (roytman) [inline kind/hack/nginx_deploy_minio.yaml:1]: Do we need to deploy the second ingress in a separate namespace? can both be deployed in the same namespace and use the same SA and RBAC roles?

File: kind/hack/tools/install_nginx.sh [hunk 3]
```diff
@@ -22,10 +24,19 @@ wait(){
 		echo "NGINX Deployment unsuccessful. Not all pods running"
 		exit $EXIT_CODE
 	fi
+
+	wait_for_pods "ingress-nginx-minio" "$MAX_RETRIES" "$SLEEP_TIME" || EXIT_CODE=$?
+
+	if [[ $EXIT_CODE -ne 0 ]]
+	then
+		echo "NGINX Deployment unsuccessful. Not all pods running"
```
Comment (roytman) [inline kind/hack/tools/install_nginx.sh]: can you change the message to something like:
""NGINX MinIO deployment unsuccessful. Not all pods running"

File: kind/hack/tools/install_nginx.sh [hunk 4]
```diff
@@ -22,10 +24,19 @@ wait(){
...
+		exit $EXIT_CODE
+	fi
 }
 
 delete(){
 	kubectl delete -f "$NGINX_INSTALLATION_FILE"
+	kubectl apply -f "$NGINX_MINIO_INSTALLATION_FILE"
```
Comment (roytman) [inline kind/hack/tools/install_nginx.sh]: kubectl **delete** -f "$NGINX_MINIO_INSTALLATION_FILE"

**Review by roytman — COMMENTED** [review]
you need to update the following files too:
```
README.md:25:> **NOTE**: Additionally, ensure that nothing is running on port 8080, which is used by the Kind cluster ingress.
README.md:36:8080 is available, creation of the cluster, installing required software (NGNIX, KubeRay and KFP), creating
README.md:124:Open the Kubeflow Pipelines UI at  http://localhost:8080/
hack/tools/kind_management.sh:33:  if lsof -Pi :8080 -sTCP:LISTEN -t >/dev/null ; then
hack/tools/kind_management.sh:34:      echo "port 8080 is in use, please clear the port and try again"
```

File: kind/hack/tools/install_nginx.sh [hunk 5]
```diff
@@ -6,11 +6,13 @@ SLEEP_TIME="${SLEEP_TIME:-50}"
 MAX_RETRIES="${MAX_RETRIES:-10}"
```
Comment (roytman) [inline kind/hack/tools/install_nginx.sh]: can you change the default at Line 5 to 60

**Review by roytman — CHANGES_REQUESTED** [review]
@D-Sai-Venkatesh thank you for your PR, several comments

File: kind/hack/populate_minio.sh [hunk 6]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
 if [ "$MINIO_SERVER" == "" ]; then
-    MINIO_SERVER="http://localhost:8080"
+    MINIO_SERVER="http://localhost:8090"
 fi
```
Comment (roytman) [inline kind/hack/populate_minio.sh:6]: Please rebase your PR, the `mc cp` path is wrong

**Review by roytman — APPROVED** [review]
LGTM

**Review by revit13 — APPROVED** [review]

**Pull request merged**